### ISSUE TITLE
z_list

### ISSUE THREAD

**Issue by ZhouCX117** [rosinality/glow-pytorch#40]
Hi, Thanks for your nice work! I am new to the glow model, so I have some stupid questions, and I don't solve them even if I try to google.

![image](https://user-images.githubusercontent.com/79554332/163681669-abfe20e0-2f01-4c03-badf-cb82f3fcd67c.png)
The flow model can translate the input $x$ to latent space code $z$ by a sequence of $h$ models. In my understanding, we only need the last output $z$ to reconstruct the input $x$. Why we don't use the learned $z$ and use a random z_list?
I appreciate your answer and hope you have a good day!

**Comment by rosinality** [comment]
If you need to reconstruct then you only need a output z lists. Random zs is used for sampling from the model.

**Comment by ZhouCX117** [comment]
Thanks for your answer! 
Why do we need a z list rather than a single z?

**Comment by rosinality** [comment]
For each block half of outputs splitted into z and appended to zs.

**Comment by ZhouCX117** [comment]
Thanks a lot! I think I have more understanding of the glow model.

**Comment by ZhouCX117** [comment]
Hi， It is me again!
I can't understand the `self.scale = nn.Parameter(torch.zeros(1, out_channel, 1, 1))` and "out = out * torch.exp(self.scale * 3)" in the ZeroConv2d module. Could you please tell me what role this plays? I  appreciate your answer!

**Comment by ZhouCX117** [comment]
`if args.n_bits < 8:
                image = torch.floor(image / 2 ** (8 - args.n_bits))

            image = image / n_bins - 0.5`
Hi, a more question. Does this code mean image=image/256-0.5? What is the good of this way?

**Comment by ZhouCX117** [comment]
`loss = -log(n_bins) * n_pixel`
Hi~ Why do we need this constant loss? I appreciate your answer! Thanks a lot!

**Comment by ZhouCX117** [comment]
Hi, do you train the face for image_size 256*256? What is the setup of hyperparameters? I set the batch=2 and keep the other, the results are so bad.

**Comment by ZhouCX117** [comment]
The parallel train can't work. And if I change the n_block from 4 to 6 as suggested by the paper, the loss becomes Nan.
